app/controllers/email_alert_subscriptions_controller.rb

```diff
@@ -25,20 +25,26 @@ def render_error(error_message)
     render action: :new
   end
 
-  def content
-    @content ||= ContentItem.from_content_store(request.path).as_hash
+  def content_item
+    @content_item ||= ContentItem.from_content_store(request.path).as_hash
   end
 
   def signup_presenter
-    @signup_presenter ||= SignupPresenter.new(content, params)
+    @signup_presenter ||= SignupPresenter.new(content_item, params)
+  end
+
+  def set_taxon
```

Comment:
Would it be possible to use the [taxon_facet](https://github.com/alphagov/finder-frontend/blob/master/app/models/taxon_facet.rb) instead of this?

(I think that's the direction we'll head in for the actual implementation, so worth exploring what re-using the existing code might look like.)

Reply:
@benthorner I refactored to reuse as much of TaxonFacet as I could. It's indeed a bit nicer than before. TaxonFacet logic is still very much tailored to dropdown selects, so that's somewhat limiting.

The next thing I want to tackle is radio button inputs with different names (`level_one_taxon` and `level_two_taxon`). On email-alert-frontend, all radio buttons have `name = "topic"`.

Reply:
Thanks for looking into that @laurentqro 👍. I would have thought that, so long as the radio inputs use those `level_` names, then everything should just work. Maybe I'm missing something subtle - that's quite likely for a new codebase.

The other thing I'm wondering is: how could this fit into the existing SignupPresenter? We likely want that to be where the code goes eventually. Maybe that can be a stretch goal - it could be more of a hindrance to the spike, than an aid to it.

Reply:
@benthorner I'll see how we could fit the logic into SignupPresenter 👌